xbmc/guilib/Texture.cpp

```diff
@@ -246,6 +246,10 @@ bool CTexture::LoadIImage(IImage* pImage,
   if (pImage->Width() == 0 || pImage->Height() == 0)
     return false;
 
+  float aspect = (float)(pImage->Width()) / pImage->Height();
+  unsigned int resolvedWidth =
```

Comment:
> In the context of textures, this would be the imageWidth. It describes a subset of the textureWidth (which might be up to 15px wider). You need to pass this imageWidth to the decoder as second parameter.

Understood.

But is it right to ignore the original aspect ratio? If we do that, the aspect ratio of the `CGUITexture` (which comes from [the skin](https://kodi.wiki/view/Aspect_Ratio)) will not work and I don't think that's what we want.
https://github.com/xbmc/xbmc/blob/4e4b2888b41d9414a270a69d38dc96c356d29fe4/xbmc/guilib/GUITexture.h#L228

Reply:
The skinning engine has multiple modes. It can keep the aspect of the texture, but it can also dictate arbitrary resolutions. I'll have to check what the best approach is, but I'm mobile ATM.

Reply:
Unfortunately, we need to go the extra mile and propagate the aspect ratio "mode" (https://kodi.wiki/view/Aspect_Ratio) through. Based on this, the actual texture resolution has to be calculated.

Right now, the current implementation reduces the texture resolution to fit the "keep" aspect ratio. "scale" and "stretch" are often taking up more screen space, but have to use the overly minified texture. "center" is totally broken, as it is scaled to "keep" sizes.